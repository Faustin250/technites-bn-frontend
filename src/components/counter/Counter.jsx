/* eslint-disable react/prop-types */
import React from "react";

const Counter = ({ animation, count }) => (
  <div className={`counter ${animation}`}>
    {count === 1 ? <p>{`${count} city`}</p> : <p>{`city ${count}`}</p>}
  </div>
);

export default Counter;
