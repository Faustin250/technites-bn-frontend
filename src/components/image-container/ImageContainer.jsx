/* eslint-disable comma-dangle */
/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
import React, { Component } from "react";
import placeholder from "../../assets/house_place_holder.jpg";

class ImageContainer extends Component {
  render() {
    return (
      <div className="image-container">
        <div>
          <img
            src={
              this.props.images[1] !== undefined
                ? this.props.images[1].images.image1
                : placeholder
            }
            alt=""
          />
        </div>
        <div>
          <img
            src={
              this.props.images[1] !== undefined
                ? this.props.images[1].images.image2
                : placeholder
            }
            alt=""
          />
        </div>
      </div>
    );
  }
}

export default ImageContainer;
