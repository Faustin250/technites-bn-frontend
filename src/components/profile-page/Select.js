import React from 'react';
import PropTypes from 'prop-types';

const Select = ({
  mode, label, name, value, options, handler, error,
}) => {
  if (mode) {
    const contents = options.map((item) => (
      <option value={item} key={item}>{item}</option>
    ));
    return (
      <div>
        <label htmlFor={name} className="m-t-5 label">
          {`${label}:`}
          <select className={error ? 'input error select' : 'input select'} value={value} name={name} onChange={handler}>
            {contents}
          </select>
        </label>
        {error ? <span style={{ color: 'red' }}>{error}</span> : null}
      </div>
    );
  }
  return (
    <div className="flex flex-column">
      <label htmlFor={name} className="m-t-5 label">{`${label}: `}</label>
      <span id={name} style={{ color: 'grey' }}>{value}</span>
    </div>
  );
};

Select.defaultProps = {
  error: '',
  value: '',
};

Select.propTypes = {
  mode: PropTypes.bool.isRequired,
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  value: PropTypes.node,
  options: PropTypes.arrayOf(PropTypes.string).isRequired,
  handler: PropTypes.func.isRequired,
  error: PropTypes.string,
};

export default Select;
