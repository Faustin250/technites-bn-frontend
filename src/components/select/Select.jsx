/* eslint-disable arrow-parens */
/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable react/prop-types */
import React from "react";

const Select = ({
  name,
  label,
  options,
  error,
  type = "text",
  myRequest,
  ...rest
}) => {
  return (
    <div className="selector">
      <label htmlFor={name}>{label}</label>
      {type === "date" ? (
        <input
          {...rest}
          type={type}
          name={name}
          disabled={myRequest ? false : true}
          className={myRequest ? "select" : "select-remove-arrow"}
        />
      ) : (
        <select
          name={name}
          id={name}
          {...rest}
          disabled={myRequest ? false : true}
          className={myRequest ? "select" : "select-remove-arrow"}
        >
          {options.map(option => (
            <option key={option.name} value={option.id}>
              {option.name}
            </option>
          ))}
        </select>
      )}
      {error && <div className="danger">{error}</div>}
    </div>
  );
};

export default Select;
