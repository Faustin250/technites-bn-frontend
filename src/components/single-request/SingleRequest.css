@import url("https://fonts.googleapis.com/css?family=Raleway&display=swap");
.image-container {
  text-align: left;
  width: 100%;
  height: 400px;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
}

.image-container div:nth-child(1) {
  width: 52%;
  height: 100%;
  border: 1px solid white;
}

.image-container div:nth-child(2) {
  width: 48%;
  height: 100%;
  border: 1px solid white;
}

.image-container div img {
  width: 100%;
  height: 100%;
  -o-object-fit: cover;
     object-fit: cover;
}

.content-container {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  padding-bottom: 2rem;
  margin: 0 auto;
  text-align: left;
  width: 78%;
  background-color: #edeff3;
}

.request-card {
  width: 100%;
  margin: 0 auto;
  border-radius: 8px;
  background-color: white;
  margin-top: 10px;
  padding: 30px;
  -webkit-box-shadow: 2px 4px 8px 0px rgba(153, 153, 153, 0.5);
          box-shadow: 2px 4px 8px 0px rgba(153, 153, 153, 0.5);
}

.description-card {
  width: 50%;
  margin: 0 auto;
  margin-top: 10px;
  padding: 10px;
  background-color: white;
  border-radius: 8px;
  -webkit-box-shadow: 2px 4px 8px 0px rgba(153, 153, 153, 0.5);
          box-shadow: 2px 4px 8px 0px rgba(153, 153, 153, 0.5);
  overflow-y: auto;
  height: 55rem;
}

.description-card input {
  width: 80%;
  margin: 20px;
  height: 200px;
  border: 1px solid grey;
  border-radius: 8px;
  outline: none;
  font-size: 15px;
  color: grey;
}

.selector label {
  font-size: 15px;
  font-family: sans-serif;
  color: #2a89f5;
  margin: 20px;
}

.select {
  border: none;
  border-bottom: 1px solid black;
  display: block;
  font-size: 16px;
  font-family: "Helvetica", "Arial", sans-serif;
  margin: 0px 20px 25px 20px;
  width: 100%;
  background: 0 0;
  text-align: left;
  color: grey;
  width: 62%;
  height: 2rem !important;
  -webkit-box-sizing: border-box;
          box-sizing: border-box;
  -webkit-appearance: none;
  -moz-appearance: none;
  border-radius: 0;
  appearance: none;
  background: url("../../assets/arrow_down.svg") no-repeat right;
}

.select:focus {
  outline-style: inherit;
}

.box {
  width: 95%;
  height: 70px;
  margin: 0 auto;
  margin-top: 10px;
  border: 1px solid black;
}

.box.wide {
  width: 95%;
  height: 200px;
  margin: 0 auto;
  margin-top: 10px;
  border: 1px solid black;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
}

.box.wide img {
  width: 50%;
  margin: 0 auto;
  height: 200px;
}

.request-description p:nth-child(2) {
  margin-left: 30px;
  color: #2c2c2c;
  font-family: "Raleway", sans-serif;
}

.request-description p:nth-child(1) {
  font-size: 1em;
  font-weight: bold;
  margin-left: 15px;
  font-family: "Raleway", sans-serif;
}

.tag {
  width: 120px;
  height: 30px;
  margin-bottom: 10px;
  border-radius: 20px;
  margin-right: 10px;
  margin-top: 20px;
  outline: none;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  color: black;
  font-size: 14px;
  -webkit-box-shadow: 2px 4px 8px 0px rgba(153, 153, 153, 0.5);
          box-shadow: 2px 4px 8px 0px rgba(153, 153, 153, 0.5);
}

.cta {
  width: 100%;
  margin-left: 20px;
}

.counter {
  width: 100px;
  height: 40px;
  border-radius: 30px 0px 0px 30px;
  background-color: #2a89f5;
  float: right;
  color: white;
  -webkit-box-shadow: 2px 6px 8px 0px rgba(153, 153, 153, 0.5);
          box-shadow: 2px 6px 8px 0px rgba(153, 153, 153, 0.5);
}

.counter p {
  text-align: center;
  font-size: 1rem;
  font-weight: bold;
  padding-top: 0.7rem;
  margin: 0;
}

.counter.animation {
  -webkit-animation: move 2s forwards alternate;
          animation: move 2s forwards alternate;
}

@-webkit-keyframes move {
  from {
    -webkit-transform: translateY(0px);
            transform: translateY(0px);
  }
  to {
    margin-right: 20%;
    -webkit-transform: translateY(220px);
            transform: translateY(220px);
  }
}

@keyframes move {
  from {
    -webkit-transform: translateY(0px);
            transform: translateY(0px);
  }
  to {
    margin-right: 20%;
    -webkit-transform: translateY(220px);
            transform: translateY(220px);
  }
}

@media only screen and (min-width: 1600px) {
  .server {
    width: 40%;
    margin: 5em auto 4em 15%;
  }
  .container {
    width: 70%;
  }
  .blue-container {
    padding: 7em 0 2em;
  }
  .blue-container ul li p,
  blockquote {
    font-size: 1.3em;
  }
  .gray-container {
    padding: 6em 0 4em;
  }
  h2 {
    margin-top: 4em;
  }
  .blue-container img {
    width: 15%;
  }
  .blue-container ul li:nth-child(3) img {
    width: 8%;
    margin-top: -4px;
  }
}

.box {
  width: 50%;
  height: 27%;
  position: absolute;
  top: 0;
  left: 0;
  opacity: 0.2;
  /* for demo purpose  */
  margin-left: 23%;
  margin-top: 39.5%;
  background: gray;
  border-radius: 8px;
  border: none;
}
/*# sourceMappingURL=SingleRequest.css.map */