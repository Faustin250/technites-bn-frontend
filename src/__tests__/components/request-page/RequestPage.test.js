/* eslint-disable comma-dangle */
/* eslint-disable import/no-named-as-default-member */
/* eslint-disable quotes */
/* eslint-disable import/named */
/* eslint-disable no-undef */
import React, { cloneElement } from "react";
import thunk from "redux-thunk";
import axios from "axios";
import moxios from "moxios";
import MockAdapter from "axios-mock-adapter";
import Enzyme, { mount, shallow } from "enzyme";
import { Route, MemoryRouter, BrowserRouter } from "react-router-dom";
import { Provider } from 'react-redux';
import configureStore from "redux-mock-store";
import Adapter from "enzyme-adapter-react-16/build";
import Select from "../../../components/select/Select";
import Counter from "../../../components/counter/Counter";
import { RequestPage, mapStateToProps as mapStateToPropsNew } from '../../../components/request-page/RequestPage';
import SingleRequest, { SingleRequest as SingleRequestComponent, mapStateToProps } from '../../../components/single-request/SingleRequest';
import ImageContainer from '../../../components/image-container/ImageContainer';

const mocks = {
  subscribe: () => jest.fn(),
  dispatch: () => jest.fn(),
  getAccomodations: () => jest.fn(),
  getRooms: () => jest.fn(),
  getUserRequests: () => jest.fn(),
};
const store = (state) => ({
  getState: () => state,
  setState: () => state,
  ...mocks
});

Enzyme.configure({ adapter: new Adapter() });

let wrapper;

describe("<request page />", () => {
  const props = {
    requests: {
      requests: [
        {
          id: 1,
          user_id: 10,
          request_type: "ReturnTrip",
          location_id: 1,
          departure_date: "2019-11-25",
          passport_name: "amily",
          passport_number: "1234567890",
          return_date: "2019-11-30",
          destinations: [
            {
              check_in: "2019-11-26",
              check_out: "2019-11-29",
              destination_id: 1,
              accomodation_id: 1
            },
            {
              check_in: "2019-11-26",
              check_out: "2019-11-29",
              destination_id: 4,
              accomodation_id: 7
            }
          ],
          reason: "visit some where jiggle",
          status: "Pending",
          createdAt: "2019-11-20T18:36:05.615Z",
          updatedAt: "2019-11-21T08:44:59.595Z"
        }
      ],
      accomodations: {
        accomodations: {
          data: {
            data: [
              {
                id: 1,
                accommodation_name: "Kigali Marriott Hotel",
                room_type: null,
                description: null,
                location: 4,
                images: null,
                quantity: null,
                services: null,
                amenities: null,
                available_space: 10,
                owner: null,
                createdAt: "2019 - 11 - 19T21: 53: 27.050Z",
                updatedAt: "2019 - 11 - 19T21: 53: 27.050Z"
              }
            ]
          }
        },
        accomodationFound: true,
        requestFound: false,
        requests: [],
        rooms: [],
      },
      rooms: {
        accomodations: {
          data: {
            data: [
              {
                id: 1,
                accommodation_name: "Kigali Marriott Hotel",
                room_type: null,
                description: null,
                location: 4,
                images: null,
                quantity: null,
                services: null,
                amenities: null,
                available_space: 10,
                owner: null,
                createdAt: "2019 - 11 - 19T21: 53: 27.050Z",
                updatedAt: "2019 - 11 - 19T21: 53: 27.050Z"
              },
              {
                id: 2,
                accommodation_name: " Kigali Serena Hotel",
                room_type: null,
                description: null,
                location: 5,
                images: null,
                quantity: null,
                services: null,
                amenities: null,
                available_space: 10,
                owner: null,
                createdAt: "2019 - 11 - 19T21: 53: 27.050Z",
                updatedAt: "2019 - 11 - 19T21: 53: 27.050Z"
              },
              {
                id: 3,
                accommodation_name: "Kigali Radisson Blu Hotel",
                room_type: null,
                description: null,
                location: 1,
                images: null,
                quantity: null,
                services: null,
                amenities: null,
                available_space: 10,
                owner: null,
                createdAt: "2019 - 11 - 19T21: 53: 27.050Z",
                updatedAt: "2019 - 11 - 19T21: 53: 27.050Z"
              },
              {
                id: 4,
                accommodation_name: "Four Seasons Hotel",
                room_type: null,
                description: null,
                location: 2,
                images: null,
                quantity: null,
                services: null,
                amenities: null,
                available_space: 10,
                owner: null,
                createdAt: "2019 - 11 - 19T21: 53: 27.050Z",
                updatedAt: "2019 - 11 - 19T21: 53: 27.050Z"
              },
            ]
          }
        },
        accomodationFound: true,
        requestFound: false,
        requests: [],
        rooms: {
          data: {
            data: [
              {
                id: 3,
                accommodation_id: 2,
                name: "Room3",
                room_type: "studio",
                description: "This is a very good room",
                cost: 100,
                images: [
                  {
                    image_url: "http://res.cloudinary.com/technites/image/upload/v1570628696/luxvju7a9rxarnsyrhwo.jpg"
                  },
                  {
                    image_url: "http://res.cloudinary.com/technites/image/upload/v1570628699/gqfczqwgjj33sdv2bmq0.jpg"
                  },
                  {
                    image_url: "http://res.cloudinary.com/technites/image/upload/v1570628701/krlafnosukdnrp3ic8nb.jpg"
                  }
                ],
                status: true,
                createdAt: "2019-11-19T21:53:27.054Z",
                updatedAt: "2019-11-19T21:53:27.054Z"
              },
              {
                id: 4,
                accommodation_id: 2,
                name: "Room4",
                room_type: "single",
                description: "This is a very good room",
                cost: 200,
                images: [
                  {
                    image_url: "http://res.cloudinary.com/technites/image/upload/v1570628696/luxvju7a9rxarnsyrhwo.jpg"
                  },
                  {
                    image_url: ""
                  },
                  {
                    image_url: "http://res.cloudinary.com/technites/image/upload/v1570628701/krlafnosukdnrp3ic8nb.jpg"
                  }
                ],
                status: true,
                createdAt: "2019-11-19T21:53:27.054Z",
                updatedAt: "2019-11-19T21:53:27.054Z"
              }
            ]
          }
        },
      }

    },
    createRequest: jest.fn(),
    deleteRequest: jest.fn(),
    editRequest: jest.fn(),
    getUserRequests: jest.fn(),
    getAccomodations: jest.fn(),
    getRooms: jest.fn(),
    match: {
      params: { id: 1 }
    },
    history: {
      replace: jest.fn()
    },
    accomodations: {
      accomodations: {
        data: {
          data: [
            {
              id: 1,
              accommodation_name: "Kigali Marriott Hotel",
              room_type: null,
              description: null,
              location: 4,
              images: null,
              quantity: null,
              services: null,
              amenities: null,
              available_space: 10,
              owner: null,
              createdAt: "2019 - 11 - 19T21: 53: 27.050Z",
              updatedAt: "2019 - 11 - 19T21: 53: 27.050Z"
            },
            {
              id: 2,
              accommodation_name: " Kigali Serena Hotel",
              room_type: null,
              description: null,
              location: 5,
              images: null,
              quantity: null,
              services: null,
              amenities: null,
              available_space: 10,
              owner: null,
              createdAt: "2019 - 11 - 19T21: 53: 27.050Z",
              updatedAt: "2019 - 11 - 19T21: 53: 27.050Z"
            },
            {
              id: 3,
              accommodation_name: "Kigali Radisson Blu Hotel",
              room_type: null,
              description: null,
              location: 1,
              images: null,
              quantity: null,
              services: null,
              amenities: null,
              available_space: 10,
              owner: null,
              createdAt: "2019 - 11 - 19T21: 53: 27.050Z",
              updatedAt: "2019 - 11 - 19T21: 53: 27.050Z"
            },
            {
              id: 4,
              accommodation_name: "Four Seasons Hotel",
              room_type: null,
              description: null,
              location: 2,
              images: null,
              quantity: null,
              services: null,
              amenities: null,
              available_space: 10,
              owner: null,
              createdAt: "2019 - 11 - 19T21: 53: 27.050Z",
              updatedAt: "2019 - 11 - 19T21: 53: 27.050Z"
            },
            {
              id: 5,
              accommodation_name: "EKO Hotels & Suites",
              room_type: null,
              description: null,
              location: 3,
              images: null,
              quantity: null,
              services: null,
              amenities: null,
              available_space: 10,
              owner: null,
              createdAt: "2019 - 11 - 19T21: 53: 27.050Z",
              updatedAt: "2019 - 11 - 19T21: 53: 27.050Z"
            },
            {
              id: 6,
              accommodation_name: "Mary Hotel",
              room_type: null,
              description: null,
              location: 3,
              images: null,
              quantity: null,
              services: null,
              amenities: null,
              available_space: 10,
              owner: null,
              createdAt: "2019 - 11 - 19T21: 53: 27.050Z",
              updatedAt: "2019 - 11 - 19T21: 53: 27.050Z"
            },
            {
              id: 7,
              accommodation_name: "New Hotel",
              room_type: null,
              description: null,
              location: 4,
              images: null,
              quantity: null,
              services: null,
              amenities: null,
              available_space: 10,
              owner: null,
              createdAt: "2019 - 11 - 19T21: 53: 27.050Z",
              updatedAt: "2019 - 11 - 19T21: 53: 27.050Z"
            },
            {
              id: 8,
              accommodation_name: "Miami Hotel",
              room_type: null,
              description: null,
              location: 5,
              images: null,
              quantity: null,
              services: null,
              amenities: null,
              available_space: 10,
              owner: null,
              createdAt: "2019 - 11 - 19T21: 53: 27.050Z",
              updatedAt: "2019 - 11 - 19T21: 53: 27.050Z"
            },
            {
              id: 9,
              accommodation_name: "Gorrilla Park",
              room_type: null,
              description: null,
              location: 1,
              images: null,
              quantity: null,
              services: null,
              amenities: null,
              available_space: 10,
              owner: null,
              createdAt: "2019 - 11 - 19T21: 53: 27.050Z",
              updatedAt: "2019 - 11 - 19T21: 53: 27.050Z"
            },
            {
              id: 10,
              accommodation_name: "Star light",
              room_type: null,
              description: null,
              location: 2,
              images: null,
              quantity: null,
              services: null,
              amenities: null,
              available_space: 10,
              owner: null,
              createdAt: "2019 - 11 - 19T21: 53: 27.050Z",
              updatedAt: "2019 - 11 - 19T21: 53: 27.050Z"
            }
          ]
        }
      },
      accomodationFound: true,
      requestFound: false,
      requests: [],
      rooms: [],
    },
    rooms: {
      accomodations: {
        data: {
          data: [
            {
              id: 1,
              accommodation_name: "Kigali Marriott Hotel",
              room_type: null,
              description: null,
              location: 4,
              images: null,
              quantity: null,
              services: null,
              amenities: null,
              available_space: 10,
              owner: null,
              createdAt: "2019 - 11 - 19T21: 53: 27.050Z",
              updatedAt: "2019 - 11 - 19T21: 53: 27.050Z"
            },
            {
              id: 2,
              accommodation_name: " Kigali Serena Hotel",
              room_type: null,
              description: null,
              location: 5,
              images: null,
              quantity: null,
              services: null,
              amenities: null,
              available_space: 10,
              owner: null,
              createdAt: "2019 - 11 - 19T21: 53: 27.050Z",
              updatedAt: "2019 - 11 - 19T21: 53: 27.050Z"
            },
            {
              id: 3,
              accommodation_name: "Kigali Radisson Blu Hotel",
              room_type: null,
              description: null,
              location: 1,
              images: null,
              quantity: null,
              services: null,
              amenities: null,
              available_space: 10,
              owner: null,
              createdAt: "2019 - 11 - 19T21: 53: 27.050Z",
              updatedAt: "2019 - 11 - 19T21: 53: 27.050Z"
            },
            {
              id: 4,
              accommodation_name: "Four Seasons Hotel",
              room_type: null,
              description: null,
              location: 2,
              images: null,
              quantity: null,
              services: null,
              amenities: null,
              available_space: 10,
              owner: null,
              createdAt: "2019 - 11 - 19T21: 53: 27.050Z",
              updatedAt: "2019 - 11 - 19T21: 53: 27.050Z"
            },
            {
              id: 5,
              accommodation_name: "EKO Hotels & Suites",
              room_type: null,
              description: null,
              location: 3,
              images: null,
              quantity: null,
              services: null,
              amenities: null,
              available_space: 10,
              owner: null,
              createdAt: "2019 - 11 - 19T21: 53: 27.050Z",
              updatedAt: "2019 - 11 - 19T21: 53: 27.050Z"
            },
            {
              id: 6,
              accommodation_name: "Mary Hotel",
              room_type: null,
              description: null,
              location: 3,
              images: null,
              quantity: null,
              services: null,
              amenities: null,
              available_space: 10,
              owner: null,
              createdAt: "2019 - 11 - 19T21: 53: 27.050Z",
              updatedAt: "2019 - 11 - 19T21: 53: 27.050Z"
            },
            {
              id: 7,
              accommodation_name: "New Hotel",
              room_type: null,
              description: null,
              location: 4,
              images: null,
              quantity: null,
              services: null,
              amenities: null,
              available_space: 10,
              owner: null,
              createdAt: "2019 - 11 - 19T21: 53: 27.050Z",
              updatedAt: "2019 - 11 - 19T21: 53: 27.050Z"
            },
            {
              id: 8,
              accommodation_name: "Miami Hotel",
              room_type: null,
              description: null,
              location: 5,
              images: null,
              quantity: null,
              services: null,
              amenities: null,
              available_space: 10,
              owner: null,
              createdAt: "2019 - 11 - 19T21: 53: 27.050Z",
              updatedAt: "2019 - 11 - 19T21: 53: 27.050Z"
            },
            {
              id: 9,
              accommodation_name: "Gorrilla Park",
              room_type: null,
              description: null,
              location: 1,
              images: null,
              quantity: null,
              services: null,
              amenities: null,
              available_space: 10,
              owner: null,
              createdAt: "2019 - 11 - 19T21: 53: 27.050Z",
              updatedAt: "2019 - 11 - 19T21: 53: 27.050Z"
            },
            {
              id: 10,
              accommodation_name: "Star light",
              room_type: null,
              description: null,
              location: 2,
              images: null,
              quantity: null,
              services: null,
              amenities: null,
              available_space: 10,
              owner: null,
              createdAt: "2019 - 11 - 19T21: 53: 27.050Z",
              updatedAt: "2019 - 11 - 19T21: 53: 27.050Z"
            }
          ]
        }
      },
      accomodationFound: true,
      requestFound: false,
      requests: [],
      rooms: {
        data: {
          data: [
            {
              id: 3,
              accommodation_id: 2,
              name: "Room3",
              room_type: "studio",
              description: "This is a very good room",
              cost: 100,
              images: [
                {
                  image_url: "http://res.cloudinary.com/technites/image/upload/v1570628696/luxvju7a9rxarnsyrhwo.jpg"
                },
                {
                  image_url: "http://res.cloudinary.com/technites/image/upload/v1570628699/gqfczqwgjj33sdv2bmq0.jpg"
                },
                {
                  image_url: "http://res.cloudinary.com/technites/image/upload/v1570628701/krlafnosukdnrp3ic8nb.jpg"
                }
              ],
              status: true,
              createdAt: "2019-11-19T21:53:27.054Z",
              updatedAt: "2019-11-19T21:53:27.054Z"
            },
            {
              id: 4,
              accommodation_id: 2,
              name: "Room4",
              room_type: "single",
              description: "This is a very good room",
              cost: 200,
              images: [
                {
                  image_url: "http://res.cloudinary.com/technites/image/upload/v1570628696/luxvju7a9rxarnsyrhwo.jpg"
                },
                {
                  image_url: ""
                },
                {
                  image_url: "http://res.cloudinary.com/technites/image/upload/v1570628701/krlafnosukdnrp3ic8nb.jpg"
                }
              ],
              status: true,
              createdAt: "2019-11-19T21:53:27.054Z",
              updatedAt: "2019-11-19T21:53:27.054Z"
            }
          ]
        }
      },
    },
    currentUser: {
      auto_fill: true
    }
  }

  const state = {
    data: { destinations: [{}] },
    errors: {},
    counter: { animation: "", count: 0 },
    cities: [
      { id: 0, name: "Choose a country" },
      { id: 1, name: "Kigali" },
      { id: 2, name: "New York" },
      { id: 3, name: "Lagos" },
      { id: 4, name: "Dubai" },
      { id: 5, name: "Kinshasa" }
    ],
    accomodations: [{ id: 0, name: "Choose an accomodation" }],
    currentAccomodations: [{ id: 0, name: "Choose an accomodation" }],
    rooms: [{ id: 0, name: "Choose a room" }],
    currentRooms: [{ id: 0, name: "Choose a room" }],
    profile: { user: { auto_fill: true } },
  };

  const initialStore = store({
    // Request: props,
    accomodations: props.requests.accomodations,
    rooms: props.requests.rooms,
    Requests: props.requests,
    profile: { user: { id: 1, firstname: 'name', lastname: "name" } }
  });

  it('it should mount the request page', () => {
    const state = mapStateToProps({ profile: { user: {} } });
    wrapper = shallow(<RequestPage {...props} />);
  });

  it('it should test for the populate request page', () => {
    wrapper = shallow(<SingleRequestComponent {...props} />);
    wrapper.setProps({ user: {} });
  });

  it('should test for the new request', () => {
    props.match.params.id = "new";
    wrapper = shallow(<RequestPage {...props} />);
  });

  it('should test for the invalid id when populating the request', () => {
    props.match.params.id = 32;
    wrapper = shallow(<RequestPage {...props} />);
  });

  it('should test for submit a request if the data is invalid', () => {
    props.match.params.id = "new";

    wrapper = shallow(<RequestPage {...props} />);
    wrapper.setState(state);
    const button = wrapper.find('button[className="button"]').first();
    button.simulate('click');

    const checkbox = wrapper.find('input[type="checkbox"]');
    checkbox.simulate('change', { target: { checked: true } });
  });

  it('should test for submit a request', () => {
    props.match.params.id = "new";

    const data = {
      request_type: 'OneWay',
      location_id: 1,
      departure_date: "2025-11-17",
      reason: "visit to cairo",
      passport_name: "some",
      passport_number: 1234567890,
      destination_id: 3,
      accomodation_id: 5,
      room_id: 9,
      check_in: "2029-09-26",
      check_out: "2029-09-27",
      destinations: [
        {
          destination_id: 3, accomodation_id: 5, room_id: 9, check_in: "2029-09-26", check_out: "2029-09-27"
        }],
    };

    wrapper = shallow(<RequestPage {...props} />);
    wrapper.setState(state);
    wrapper.setState({ data });
    const button = wrapper.find('button[className="button"]').first();
    button.simulate('click');
  });

  it('should test for submit a multi city request', () => {
    props.match.params.id = "new";

    const data = {
      request_type: 'OneWay',
      location_id: 1,
      departure_date: "2025-11-17",
      reason: "visit to cairo",
      passport_name: "some",
      passport_number: 1234567890,
      destination_id: 3,
      accomodation_id: 5,
      room_id: 9,
      check_in: "2029-09-26",
      check_out: "2029-09-27",
      destinations: [
        {
          destination_id: 3, accomodation_id: 5, room_id: 9, check_in: "2029-09-26", check_out: "2029-09-27"
        }],
    };

    wrapper = shallow(<RequestPage {...props} />);
    wrapper.setState(state);
    wrapper.setState({ data });
    const button = wrapper.find('button[className="button"]').at(1);
    button.simulate('click');
  });

  it('should test for editing a request', () => {
    props.match.params.id = 1;

    const data = {
      request_type: 'OneWay',
      location_id: 1,
      departure_date: "2025-11-17",
      reason: "visit to cairo",
      passport_name: "some",
      passport_number: 1234567890,
      destination_id: 3,
      accomodation_id: 5,
      room_id: 9,
      check_in: "2029-09-26",
      check_out: "2029-09-27",
      destinations: [
        {
          destination_id: 3, accomodation_id: 5, room_id: 9, check_in: "2029-09-26", check_out: "2029-09-27"
        }],
    };

    wrapper = shallow(<RequestPage {...props} />);
    wrapper.setState(state);
    wrapper.setState({ data });
    const button = wrapper.find('button[className="button"]').first();
    button.simulate('click');
  });

  it('should test for delete a request', () => {
    props.match.params.id = 1;

    const data = {
      request_type: 'OneWay',
      location_id: 1,
      departure_date: "2025-11-17",
      reason: "visit to cairo",
      passport_name: "some",
      passport_number: 1234567890,
      destination_id: 3,
      accomodation_id: 5,
      room_id: 9,
      check_in: "2029-09-26",
      check_out: "2029-09-27",
      destinations: [
        {
          destination_id: 3, accomodation_id: 5, room_id: 9, check_in: "2029-09-26", check_out: "2029-09-27"
        }],
    };

    wrapper = shallow(<RequestPage {...props} />);
    wrapper.setState(state);
    wrapper.setState({ data });
    const button = wrapper.find('button[className="button"]').at(1);
    button.simulate('click');
  });

  it('should test for back button', () => {
    props.match.params.id = 1;

    const data = {
      request_type: 'OneWay',
      location_id: 1,
      departure_date: "2025-11-17",
      reason: "visit to cairo",
      passport_name: "some",
      passport_number: 1234567890,
      destination_id: 3,
      accomodation_id: 5,
      room_id: 9,
      check_in: "2029-09-26",
      check_out: "2029-09-27",
      destinations: [
        {
          destination_id: 3, accomodation_id: 5, room_id: 9, check_in: "2029-09-26", check_out: "2029-09-27"
        }],
    };

    wrapper = shallow(<RequestPage {...props} />);
    wrapper.setState(state);
    wrapper.setState({ data });
    state.counter.count = 1;
    wrapper.setState(state);
    const button = wrapper.find('button[className="button"]').first();
    button.simulate('click');
  });

  it('should test for forward button', () => {
    props.match.params.id = 1;

    const data = {
      request_type: 'OneWay',
      location_id: 1,
      departure_date: "2025-11-17",
      reason: "visit to cairo",
      passport_name: "some",
      passport_number: 1234567890,
      destination_id: 3,
      accomodation_id: 5,
      room_id: 9,
      check_in: "2029-09-26",
      check_out: "2029-09-27",
      destinations: [
        {
          destination_id: 3, accomodation_id: 5, room_id: 9, check_in: "2029-09-26", check_out: "2029-09-27"
        }],
    };

    wrapper = shallow(<RequestPage {...props} />);
    wrapper.setState(state);
    wrapper.setState({ data });
    state.counter.count = 1;

    state.data.destinations.push(
      {
        destination_id: 3, accomodation_id: 5, room_id: 9, check_in: "2029-09-26", check_out: "2029-09-27"
      }
    );
    state.data.destinations.push(
      {
        destination_id: 3, accomodation_id: 5, room_id: 9, check_in: "2029-09-26", check_out: "2029-09-27"
      }
    );
    wrapper.setState(state);
    const button = wrapper.find('button[className="button"]').at(1);
    button.simulate('click');
  });

  it('should test for setting state to props in request page', () => {
    // mapStateToPropsNew({ Requests: {}, accomodations: {}, rooms: {}, currentUser: { profile: { user: { auto_fill: true } } } });
    mapStateToPropsNew(state);
  });

  it('should test select element', () => {
    const selectProps = {
      name: '',
      value: '',
      label: '',
      type: 'text',
      options: [{ id: 1, name: '' }],
      onChange: jest.fn(),
      error: '',
    };

    shallow(<Select {...selectProps} />);
  });

  it('should test counter element', () => {
    const counterProps = {
      animation: 'animation',
      count: 0,
    };

    shallow(<Counter {...counterProps} />);
  });

  it("should check if the input field are being rendered", () => {
    const destinations = [
      { id: 0, name: "Choose a country" },
      { id: 1, name: "Kigali" },
    ];

    const accomodations = [
      { id: 0, name: "Choose an accomodation" },
      { id: 1, name: "Hotel" },
    ];

    const requestPage = mount(<RequestPage {...props} />);
    const destinationSelect = requestPage.find('select[name="destination_id"]');
    destinationSelect.simulate('change', destinations[1]);

    const accomodationSelect = requestPage.find('select[name="accomodation_id"]');
    accomodationSelect.simulate('change', accomodations[1]);
  });

  it("should test the image container", () => {
    const imageProps = {
      images: [
        { images: { image1: '', image2: '' } },
        { images: { image1: '', image2: '' } },
      ]
    };

    const requestPage = mount(<ImageContainer {...imageProps} />);
  });
});
