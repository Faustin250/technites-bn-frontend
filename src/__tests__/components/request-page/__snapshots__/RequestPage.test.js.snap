// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<request page /> it should mount the request page 1`] = `ShallowWrapper {}`;

exports[`<request page /> it should mount the request page as a travel admin logged in 1`] = `ShallowWrapper {}`;

exports[`<request page /> it should test for the populate request page 1`] = `ShallowWrapper {}`;

exports[`<request page /> it should test for the populate request page with props new 1`] = `ShallowWrapper {}`;

exports[`<request page /> should test for the new request 1`] = `ShallowWrapper {}`;
